lib/Table.js

```diff
@@ -515,56 +544,71 @@ const buildTableReq = Table.prototype.buildTableReq = function buildTableReq(nam
 };
 
 Table.prototype.create = function (next) {
-  let ddb = this.base.ddb();
-  let schema = this.schema;
-  let createTableReq = this.buildTableReq(this.name, schema);
 
-  debug('ddb.createTable request:', createTableReq);
+  if(next){
+    this.create()
+      .then(result => {next(undefined,result)}, err => {next(err)})
+    return
+  }
 
-  const deferred = Q.defer();
+  return new Promise((resolve, reject) => {
+    let ddb = this.base.ddb();
+    let schema = this.schema;
+    let createTableReq = this.buildTableReq(this.name, schema);
 
-  ddb.createTable(createTableReq, function (err, data) {
-    if (err) {
-      debug('error creating table', err);
-      return deferred.reject(err);
-    }
-    debug('table created', data);
-    deferred.resolve(data);
-  });
-  return deferred.promise.nodeify(next);
+    debug('ddb.createTable request:', createTableReq);
 
+
+    ddb.createTable(createTableReq, function (err, data) {
+      if (err) {
+        debug('error creating table', err);
+        return reject(err);
+      }
+      debug('table created', data);
+      resolve(data);
+    });
+  })
 };
 
 Table.prototype.delete = function (next) {
   let deleteTableReq = {
     TableName: this.name
   };
 
-  debug('ddb.deleteTable request:', deleteTableReq);
-
-  let ddb = this.base.ddb();
+  if(next){
```

Comment:
See comment above about this

Reply:
This should be taken out in the latest changes.